R/db.R

```diff
@@ -203,19 +215,92 @@ setMethod("sqlCreateTable", "DB2/AIX64",
 
 # Microsoft SQL Server ---------------------------------------------------------
 
-# Simple class prototype to avoid messages about unknown classes from setMethod
+#' Simple class prototype to avoid messages about unknown classes from setMethod
+#' @rdname SQLServer
+#' @usage NULL
 setClass("Microsoft SQL Server", where = class_cache)
 
-# For SQL Server, conn@quote will return the quotation mark, however
-# both quotation marks as well as square bracket are used interchangeably for
-# delimited identifiers.  See:
-# https://learn.microsoft.com/en-us/sql/relational-databases/databases/database-identifiers?view=sql-server-ver16
-# Therefore strip the brackets first, and then call the DBI method that strips
-# the quotation marks.
-# TODO: the generic implementation in DBI should take a quote char as
-# parameter.
+#' SQL Server specific implementation.
+#'
+#' For SQL Server, conn@quote will return the quotation mark, however
+#' both quotation marks as well as square bracket are used interchangeably for
+#' delimited identifiers.  See:
+#' \url{https://learn.microsoft.com/en-us/sql/relational-databases/databases/database-identifiers?view=sql-server-ver16}
+#' Therefore strip the brackets first, and then call the DBI method that strips
+#' the quotation marks.
+#' TODO: the generic implementation in DBI should take a quote char as
+#' parameter.
+#'
+#' @rdname SQLServer
+#' @docType methods
+#' @inheritParams DBI::dbUnquoteIdentifier
+#' @usage NULL
 setMethod("dbUnquoteIdentifier", c("Microsoft SQL Server", "SQL"),
   function(conn, x, ...) {
     x <- gsub("(\\[)([^\\.]+?)(\\])", "\\2", x)
     callNextMethod( conn, x, ... )
   })
+
+#' SQL Server specific implementation.
+#'
+#' Local temp tables are stored as
+#' \code{ [tempdb].[dbo].[#name]________(padding using underscores)[numeric identifier] }
+#'
+#' True if:
+#' - If catalog_name is supplied it must equal "temdb" or "%" ( wildcard )
+#' - Name must start with "#" followd by a non-"#" character
```

Comment:
I'd suggest just writing all these method docs in a bulleted list in the `SQLServer` docs. That makes it a bit easier to see how the final rendered doc will look.

And I think it's fine to leave the `@usage` as is, so the read can more easily see what methods are customised.

Reply:
Thanks Hadley.

I took a look at the markdown in `SQLServer.md` and fwict it already looks like a bulleted list:

```
  \details{
  True if:
  \itemize{
  \item If catalog_name is supplied it must equal "temdb" or "\%" ( wildcard )
  \item Name must start with "#" followd by a non-"#" character
  }
```
Let me know if you had something else in mind.

RE: usage, I spend some time struggling with this:

```
❯ checking for code/documentation mismatches ... WARNING
  Functions or methods with usage in documentation object 'Microsoft SQL Server-class' but not in code:
    ‘\S4method{isTempTable}{Microsoft SQL Server,character}’
```

My first thought was that it has something to do with the class name having white-space characters but now i am not so sure.

The class and method are tagged like so:

```
  \name{Microsoft SQL Server-class}                                                                                                                                                                                                         
  \alias{Microsoft SQL Server-class}
  \alias{isTempTable,Microsoft SQL Server,character-method}
```
while the usage section:
```
\usage{
  \S4method{isTempTable}{`Microsoft SQL Server`,character}(conn, name, catalog_name = NULL, schema_name = NULL, ...)                                                                     
}
```

Let me know if you have any ideas.

Thanks again!

Reply:
If you manually change the alias to the following, does the problem go away?

```
  \alias{isTempTable,`Microsoft SQL Server`,character-method}
```

Reply:
Good suggestion; alas doesn't work.

Reply:
Weird! Definitely not worth looking into further.